app/models/supporter.rb

```diff
@@ -135,12 +138,18 @@ def address_field_attributes
     attributes.slice(*ADDRESS_FIELDS)
   end
 
+  def filled_address_fields?
+    address_field_attributes.any? { |column, value| value.present? }
+  end
+
   def update_primary_address
     if self.changes.slice(*ADDRESS_FIELDS).any? #changed an address field
-      unless primary_address.nil?
-        primary_address.update(address_field_attributes)
+      if primary_address.nil?
+        if filled_address_fields?
+          self.addresses.build(address_field_attributes)
+        end
       else
-        self.addresses.build(address_field_attributes)
+        primary_address.update(address_field_attributes)
```

Comment:
If primary_address exists and then the fields passed in are no longer filled, do we want to keep the primary address or delete it? I would say delete it but I'm not sure.

Reply:
I think if all fields are nil, then we could delete it. What happens if the supporter has other address? It becomes the primary address? Or nothing happens?

Reply:
It should be reassigned to another primary address. However, that's not exactly implemented right now.

Reply:
If you look at line 69, it shows that we assign on "after_add" but we don't on "after_remove". Going forward, we should but it wasn't implemented probably because there's no need to do that if a supporter only has a single address (which we assume to still be true)

Reply:
I was working on deleting the primary address when the primary address is being filled with nil values but I was not able to delete it because of the association. I thought about doing the logical deletion but I don't think it makes sense to keep records that are only made of nil values.

I'm getting a 
```
RuntimeError:
       Can't modify frozen hash
```

This is what I'm trying to do: https://github.com/CommitChange/houdini/pull/140/files#diff-f53f9da246e7dd1193ae67566477a021433a6c4bd8cb3abb7ab15496a0695103R154-R156

I also tried other combinations, like:

```ruby
        prim_addr = primary_address
        self.addresses.delete(primary_address)
        prim_addr.supporter.delete
        primary_address.delete
        prim_addr.destroy
```

Can you help with this, @wwahammy?